webhooks/virtualmachine/validation/virtualmachine_validator.go

```diff
@@ -1172,15 +1172,17 @@ func (v validator) validateVolumes(
 			allErrs = append(allErrs, field.Required(volPath.Child("persistentVolumeClaim"), ""))
 		} else {
 			allErrs = append(allErrs,
-				v.validateVolumeWithPVC(ctx, oldVolumesMap[vol.Name], vol, volPath)...)
+				v.validateVolumeWithPVC(ctx, oldVM, vm, oldVolumesMap[vol.Name], vol, volPath)...)
 		}
 	}
 
 	return allErrs
 }
 
 func (v validator) validateVolumeWithPVC(
-	_ *pkgctx.WebhookRequestContext,
+	ctx *pkgctx.WebhookRequestContext,
+	oldVM *vmopv1.VirtualMachine,
```

Comment:
It looks like the convention used elsewhere `vm, oldVM`. You shouldn't need to check if `vm` is nil